### PR TITLE
refactor(node): send proof-chain only for specific messages

### PR DESCRIPTION
**Pull request by Yoga07**

### CODE REVIEW

**Comment by lionel-faber** [comment]
Needs `BREAKING CHANGE`

File: src/node/mod.rs [hunk 1]
```diff
@@ -57,8 +56,8 @@ pub struct RoutingMsg {
     pub aggregation: Aggregation,
     /// The body of the message.
     pub variant: Variant,
-    /// Proof chain to verify the message trust. Does not need to be signed.
-    pub proof_chain: Option<SecuredLinkedList>,
+    /// Section key of the sender.
+    pub section_pk: threshold_crypto::PublicKey,
```
Comment (maqi) [inline src/node/mod.rs]: you can use `BlsPublicKey` directly

File: src/node/mod.rs [hunk 2]
```diff
@@ -84,6 +83,10 @@ impl RoutingMsg {
     pub fn serialize(&self, dest: XorName, dest_section_pk: BlsPublicKey) -> crate::Result<Bytes> {
         WireMsg::serialize_routing_msg(self, dest, dest_section_pk)
     }
+
+    pub fn section_pk(&self) -> threshold_crypto::PublicKey {
```
Comment (maqi) [inline src/node/mod.rs]: same here, use `BlsPublicKey` directly

File: src/node/mod.rs [hunk 3]
```diff
@@ -92,7 +95,6 @@ impl PartialEq for RoutingMsg {
             && self.dst == other.dst
             && self.id == other.id
             && self.variant == other.variant
-            && self.proof_chain == other.proof_chain
```
Comment (maqi) [inline src/node/mod.rs:95]: do we need to check `section_pk` is equal as well?

**Review by maqi — COMMENTED** [review]
also, just to confirm that the `BREAKING CHANGE` is included in the commit description.
it will be better to name the commit titile as `api!` something as well.

**Review by maqi — APPROVED** [review]

**Pull request merged**